### ISSUE TITLE
Unexpected keyword 'yield' error in transpiled code.

### ISSUE THREAD

**Issue by ernestostifano** [babel/babel#11443]
## Bug Report

- [ ] I would like to work on a fix!

**Current Behavior**

Transpiled code throws `SyntaxError: Unexpected strict mode reserved word` on runtime.

Static analysis complains `Unexpected keyword 'yield'. Cannot use yield expression out of generator.`

**Input Code**
- REPL or Repo link if applicable: [REPL Link Here](https://babeljs.io/repl#?browsers=since%202015&build=&builtIns=false&spec=true&loose=false&code_lz=JYWwDg9gTgLgBAbwjAFgUygMQK4DsDGMwEuAvnAGZQQhwDkAdAPQDONaTy6UAtBXoWK4GAKxZ0A3ACgpAQxYBPApQFEScNiDQ4Ca3AApZUAOYBKRFLhW4sgO6zg8UJFj7L1gJB2H8Lhh2CJPqm7lam0qQyaAAeLvAImtqqQqQSQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env%2Cenv&prettier=false&targets=Node-10&version=7.9.0&externalPlugins=)

```js
import {otherFunction} from './some/other-function.js';

async function someFunction(arg) {
    await import(
    	await otherFunction()
    );
}

export {someFunction};
```

**Expected behavior/code**

No error to be present.

**Babel Configuration (babel.config.js, .babelrc, package.json#babel, cli command, .eslintrc)**
- Filename: `babel.config.js`

```js
{
  presets: [
    [
        '@babel/preset-env',
        {
            spec: true,
            targets: {
                browsers: 'since 2015',
                node: '10.0.0'
            },
            ignoreBrowserslistConfig: true
        }
    ]
  ]
}
```

**Environment**

- Babel version(s): 7.9.0
- Node/npm version: Node 13.12.0 / NPM 6.14.4
- OS: macOS Catalina 10.15.3
- Monorepo: Lerna
- How you are using Babel: cli

**Possible Solution**

N/A.

**Additional context/Screenshots**

N/A.

**Comment by nicolo-ribaudo** [comment]
@ernestostifano I have re-opened this issue in the repository we internally use to transform dynamic import: https://github.com/airbnb/babel-plugin-dynamic-import-node/issues/84

**Comment by JLHwung** [comment]
@nicolo-ribaudo Is it fixed by #11458?

**Comment by nicolo-ribaudo** [comment]
Yes!
